sulong/projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/pointer/CommonPointerLibraries.java

```diff
@@ -116,6 +143,79 @@ static boolean isMemberModifiable(LLVMPointerImpl receiver, String ident,
         }
     }
 
+    @ExportMessage
+    static boolean isMemberInvocable(LLVMPointerImpl receiver, String ident) {
+        LLVMInteropType type = receiver.getExportType();
+        if (type instanceof LLVMInteropType.Clazz) {
+            LLVMInteropType.Clazz clazz = (LLVMInteropType.Clazz) type;
+            return clazz.findMethod(ident) != null;
+        }
+        return false;
+    }
+
+    @ExportMessage
+    static class InvokeMember {
+        @SuppressWarnings("unused")
+        @Specialization(guards = {"asClazz(receiver)==clazz", "member.equals(methodName)", "argCount==arguments.length"})
+        static Object doCached(LLVMPointerImpl receiver, String member, Object[] arguments,
+                        @CachedContext(LLVMLanguage.class) LLVMContext context, @CachedLibrary(limit = "5") InteropLibrary interop,
+                        @Cached(value = "asClazz(receiver)") LLVMInteropType.Clazz clazz,
+                        @Cached(value = "clazz.findMethodByArguments(receiver, member, arguments)") Method method,
+                        @Cached(value = "arguments.length") int argCount,
+                        @Cached(value = "method.getName()") String methodName,
+                        @Cached(value = "getLLVMFunction(context, method, clazz, member)") LLVMFunction llvmFunction,
+                        @Cached(value = "create(llvmFunction)") LLVMAccessSymbolNode accessSymbolNode)
+                        throws UnsupportedMessageException, ArityException, UnsupportedTypeException {
+            Object[] newArguments = addSelfObject(receiver, arguments);
+            return interop.execute(accessSymbolNode.execute(), newArguments);
+        }
+
+        @SuppressWarnings("unused")
+        @Specialization(replaces = "doCached")
+        static Object doResolve(LLVMPointerImpl receiver, String member, Object[] arguments,
+                        @CachedContext(LLVMLanguage.class) LLVMContext context, @CachedLibrary(limit = "5") InteropLibrary interop,
+                        @Cached(value = "create()", allowUncached = true) LLVMDynAccessSymbolNode dynAccessSymbolNode)
```

Comment:
`allowUncached` for nodes is wrong, that would create a new node for every call when used through an uncached library. Put the `@GenerateUncached` annotation on the `LLVMDynAccessSymbolNode`, then it should work without `allowUncached = true`. Also `value = "create()"` shouldn't be necessary, since that's the default (but not wrong either).